src/Application.php

```diff
@@ -25,6 +29,48 @@ public function getTransformationConfig(int $transformationId): array
         return $this->removeDisableTransformation($transformationConfig);
     }
 
+    public function migrateTransformationConfig(array $transformationConfig): array
+    {
+        $transformationsV2 = [];
+        foreach ($transformationConfig['rows'] as $row) {
+            $transformationKey = sprintf(
+                '%s-%s',
+                $row['configuration']['backend'],
+                $row['configuration']['type']
+            );
+
+            $transformationV2 = new TransformationV2(
```

Comment:
@ondrajodas 
Ak spravne chapem kodu, tak pre kazdy `row` z konfiguracie vytvaras novu v2 transformaciu. Je to tak?

Mas sice metodu `processRow`, ktora robi aj nejake mergovanie IM/OM, ale vzdy sa vola iba raz per row, takze IMHO nic nemerguje.

Podla Jiry by to malo fungovat inak:

```
each transformation bucket = 1 new transformation

if all transformation have the same backend

merge transformations mapping

check if there is only one phase

```

:man_shrugging: 
Teda nevim, ci tomu zle chapem, alebo sa menilo zadanie, ... vysvetlis mi to prosim?

Reply:
koukni na test:
https://github.com/keboola/app-transformation-migrate/tree/COM-663-ondra-migrate-same-backend/tests/functional/migrate-transformations-with-storage

kde vytvářím dvě transformace:
https://github.com/keboola/app-transformation-migrate/blob/COM-663-ondra-migrate-same-backend/tests/functional/migrate-transformations-with-storage/setUp.php#L12

https://github.com/keboola/app-transformation-migrate/blob/COM-663-ondra-migrate-same-backend/tests/functional/migrate-transformations-with-storage/setUp.php#L43

ze kterých se vytvoří jedná nová s mergnutými IM/OM:
https://github.com/keboola/app-transformation-migrate/blob/COM-663-ondra-migrate-same-backend/tests/functional/migrate-transformations-with-storage/expected/data/out/transformationDump/transformations.json

Reply:
ten `transformationKey` který tu vytvářím:
https://github.com/keboola/app-transformation-migrate/blob/COM-663-ondra-migrate-same-backend/src/Application.php#L36

je pravda možná trochu zbytečný, protože je tady v tom případě je vždy stejný a načte se jedna a ta samá nová transformace, ale je to spíš příprava na další upgrade kdy už ty backendy nemusí být stejný a bude se vytvářet více transformací

Reply:
ok vdaka, idem sa na to pozriet

Reply:
@webrouse ^

Reply:
když budeš chtít tak si můžeme zavolat a vysvětlím víc

Reply:
@webrouse možná ještě na příkladu:
mám tu transformaci v1 - https://connection.keboola.com/admin/projects/6231/transformations/bucket/639113714
která je po zmigrování - https://connection.keboola.com/admin/projects/6231/transformations-v2/keboola.snowflake-transformation/687015450